webapp/src/blocks/archive.ts

```diff
@@ -2,10 +2,80 @@
 // See LICENSE.txt for license information.
 import {IBlock} from './block'
 
-interface IArchive {
+interface IArchiveHeader {
     version: number
     date: number
-    blocks: readonly IBlock[]
 }
 
-export {IArchive}
+interface IArchiveLine {
+    type: string,
+    data: any,
+}
+
+// This schema allows the expansion of additional line types in the future
+interface IBlockArchiveLine extends IArchiveLine {
+    type: 'block',
+    data: IBlock
+}
+
+class ArchiveUtils {
+    static buildBlockArchive(blocks: readonly IBlock[]): string {
+        const header: IArchiveHeader = {
+            version: 1,
+            date: Date.now(),
+        }
+
+        const headerString = JSON.stringify(header)
+        let content = headerString + '\n'
+        for (const block of blocks) {
+            const line: IBlockArchiveLine = {
+                type: 'block',
+                data: block,
+            }
+            const lineString = JSON.stringify(line)
```

Comment:
JSON.stringify in Chrome outputs a single-line by default, although it's not clear from [the spec](https://tc39.es/ecma262/#sec-json.stringify) if this is true for all JS engines. May need to keep an eye on this.